Description: break dependency loop by making source-map module optional
Forwarded: https://github.com/mishoo/UglifyJS2/pull/477
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2014-05-02

--- a/package.json
+++ b/package.json
@@ -16,10 +16,12 @@
     },
     "dependencies": {
         "async"      : "~0.2.6",
-        "source-map" : "0.1.34",
         "optimist"   : "~0.3.5",
         "uglify-to-browserify": "~1.0.0"
     },
+    "optionalDependencies": {
+        "source-map" : "0.1.34"
+    },
     "browserify": {
         "transform": [ "uglify-to-browserify" ]
     },
--- a/tools/node.js
+++ b/tools/node.js
@@ -2,11 +2,16 @@
 var fs = require("fs");
 var vm = require("vm");
 var sys = require("util");
+var sourceMap;
+try {
+	sourceMap = require("source-map");
+} catch(e) {
+}
 
 var UglifyJS = vm.createContext({
     sys           : sys,
     console       : console,
-    MOZ_SourceMap : require("source-map")
+    MOZ_SourceMap : sourceMap
 });
 
 function load_global(file) {
@@ -107,17 +112,21 @@
         inMap = fs.readFileSync(options.inSourceMap, "utf8");
     }
     if (options.outSourceMap) {
-        output.source_map = UglifyJS.SourceMap({
-            file: options.outSourceMap,
-            orig: inMap,
-            root: options.sourceRoot
-        });
-        if (options.sourceMapIncludeSources) {
-            for (var file in sourcesContent) {
-                if (sourcesContent.hasOwnProperty(file)) {
-                    output.source_map.get().setSourceContent(file, sourcesContent[file]);
+        if (sourceMap) {
+            output.source_map = UglifyJS.SourceMap({
+                file: options.outSourceMap,
+                orig: inMap,
+                root: options.sourceRoot
+            });
+            if (options.sourceMapIncludeSources) {
+                for (var file in sourcesContent) {
+                    if (sourcesContent.hasOwnProperty(file)) {
+                        output.source_map.get().setSourceContent(file, sourcesContent[file]);
+                    }
                 }
             }
+        } else {
+            console.error("source-map module is missing and needed by outSourceMap option");
         }
 
     }
